ratel/src/ast/expression.rs

```diff
@@ -216,4 +216,19 @@ impl<'ast> Expression<'ast> {
             _           => true,
         }
     }
+
+    #[inline]
+    pub fn is_lvalue(&self) -> bool {
+        use self::Expression::*;
+
+        match *self {
+            Expression::Identifier(_) |
+            Expression::Member(_)     |
+            Expression::Object(_)     |
+            Expression::Array(_)      |
+            Expression::Spread(_)     => true,
```

Comment:
`Identifier`, `Member` and `ComputedMember` I think are the obvious ones here, but will double check later. The rest is contextually correct (Array and Object is valid if they are using as deconstruction patterns, we don't check that yet).